trismart_smartboard_integration/models/smartboard_connector.py

```diff
@@ -2,41 +2,47 @@
+            if sb_lead_id and project_template_id:
+                partner_exist = self.env['res.partner'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                lead_exist = self.env['crm.lead'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                project_exist = self.env['project.project'].search([('sb_lead_id', '=', sb_lead_id)], limit=1)
+                partner_id = []
 
                 if not partner_exist:
                     # Create new partner
-                    partner_id = self.env['res.partner'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0], "sync_status" : '1'})
+                    new_id = self.env['res.partner'].create({'name': 'SBLead-' + sb_lead_id, 'sb_lead_id': sb_lead_id, "sync_status" : '1'})
+                    partner_id.append(new_id)
+
                 if not lead_exist:
                     # Create new lead
-                    lead_id = self.env['crm.lead'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0]})
-
+                    lead_id = self.env['crm.lead'].create({'name': 'SBLead-' + sb_lead_id, 'sb_lead_id': sb_lead_id})
                     # Convert lead to opportunity
-                    customer = self.env['res.partner'].browse(partner_id) if partner_id else partner_exist
-                    for lead in lead_id:
-                        vals = lead._convert_opportunity_data(customer, team_id=False)
-                        lead.write(vals)
+                    if partner_id:
+                        customer = self.env['res.partner'].browse(partner_id[0])
+                    else:
+                        customer = self.env['res.partner'].browse(partner_exist)
+                    vals = lead_id[0]._convert_opportunity_data(customer, team_id=False)
+                    lead_id[0].write(vals)
 
                 if not project_exist:
                     # Create new project
-                    project_id = self.env['project.project'].create({'name': 'SBLead-' + sb_lead_id[0], 'sb_lead_id': sb_lead_id[0]})
-                    project_id.partner_id = partner_id
+                    new_project = self.env['project.project'].search([('id', '=', project_template_id)]).create_project_from_template(sb_lead_id)
+                    project = self.env['project.project'].search([('id', '=', new_project['res_id'])])
+                    if partner_id:
+                        project.partner_id = partner_id[0].id
+                    else:
+                        project.partner_id = partner_exist.id
 
-                if partner_exist:
-                    return {"status": 200,  "message": "partner exists", "odoo_customer_id": []}
-                else:
-                    return {"status": 200, "message": "partner created", "odoo_customer_id": partner_id.id}
+                if not partner_exist:
+                    return {"status": 200, "error": "partner created", "odoo_customer_id": partner_id[0].id}
```

Comment:
Is it weird when the status is 200 and we have error message?

Reply:
resolved in v3